CHANGES.rst

```diff
@@ -1,3 +1,7 @@
+Upcoming version
+================
+- The adapter and instrument mechanics have been overhauled. Nothing should have changed for the user.
```

Comment:
What we _should_ document, though, is all the changes for instrument implementors.
Off my (by now) hazy recollection: The fate of `query_delay`, instrument-level `preprocess_reply`, mention new convention for which methods to override (point to docs), etc. (you surely have a better picture at this point)